### DISCUSSION TITLE
use `pydantic` for defining schemas?

### DISCUSSION THREAD

**Discussion by dyf** [AllenNeuralDynamics/data_schema#72] [Ideas]
`pydantic` is an interesting library for defining schema with well-supported converters to go back and forth from JSONSchema. Should we write schemas in `pydantic` instead of raw JSONschema?

draft PR where I am exploring this: https://github.com/AllenNeuralDynamics/data_schema/pull/63

@saskiad @jtyoung84

**Comment by jtyoung84** [comment]
Yeah. It will make it easier to import it into other projects. We can also autogenerate the json and database them on the cloud. Pydantic probably makes things easier to maintain, although protobuf might be an interesting alternative.
